src/shogun-core-main/src/test/java/de/terrestris/shoguncore/web/MapControllerTest.java

```diff
@@ -61,14 +61,9 @@ public void setLayersForMapTest_erroneous() throws Exception {
         MvcResult result = mockMvc.perform(MockMvcRequestBuilders.post(SETLAYERS_ENDPOINT)
                 .param("mapModuleId", "")
                 .param("layerIds", ""))
-                .andExpect(status().isOk()).andReturn();
-
-        assertEquals(result.getResponse().getStatus(), 200);
+                .andExpect(status().is4xxClientError()).andReturn();
```

Comment:
This looks a bit as if the API has changed after the upgrades. Is this correct? Would this require a major release?

Reply:
It depends on how you look at it. Due to the Spring upgrade, Spring reacts differently if you send it an empty string to a place where an actual mandatory parameter is expected. Previously you'd get a `200` with whatever response the controller produces in this case, now you get a `400` since Spring detects this before giving up control to the controller.

The new behaviour is actually more consistent how Spring works in general, e.g. if you send it a string where a number is expected you'll also get a `400` with the old version.

Now to discuss whether this changes the API, you can surely argue that it does. You could also argue that this actually changes a bug in the API. I prefer to think of the old behaviour as a bug, we're not using Spring to annotate the controller with the required/optional parameters, the allowed content-type, the required HTTP method etc. just to check if all these requirements are met ourselves.

I'll leave it to the next person to make a release to think more about this.

Reply:
Thanks for the clarification. I'd still think it is a major one. Let's dicuss this then.

How do we ensure we do not forget about this discussion? Adding a label to this issue?

Reply:
I've opened a [discussion](https://github.com/terrestris/shogun-core/discussions/413)